packages/notifications/push/amplify_push_notifications/android/src/main/kotlin/com/amazonaws/amplify/amplify_push_notifications/PushNotificationBackgroundService.kt

```diff
@@ -0,0 +1,169 @@
+     */
+    private var backgroundFlutterEngine: FlutterEngine? = null
+
+    /**
+     * Indicates starting of the background Flutter engine
+     */
+    private val serviceStarted = AtomicBoolean(false)
+
+    /**
+     * The Background Channel that is used to send messages to the callback dispatcher
+     */
+    private lateinit var backgroundChannel: MethodChannel
+
+    companion object {
+        private val JOB_ID = UUID.randomUUID().mostSignificantBits.toInt()
+        fun enqueueWork(context: Context, work: Intent) {
+            enqueueWork(context, PushNotificationBackgroundService::class.java, JOB_ID, work)
+        }
+    }
+
+    override fun onCreate() {
+        super.onCreate()
+        startPushNotificationService(this)
+    }
+
+    private fun startPushNotificationService(context: Context) {
+        if (backgroundFlutterEngine != null) {
+            return
+        }
+        try {
+
+            Log.i(TAG, "Starting PushNotificationBackgroundService")
+
+            synchronized(serviceStarted) {
+                val mainHandler = Handler(context.mainLooper)
+                mainHandler.post {
+                    val loader = FlutterLoader()
+                    loader.startInitialization(context)
+                    loader.ensureInitializationCompleteAsync(
+                        context,
+                        null,
+                        mainHandler,
+                    ) {
+                        createAndRunBackgroundEngine(context, loader)
+                    }
+                }
+            }
+        } catch (exception: Exception) {
+            Log.e(TAG, "Fatal error retrieving background processor info: $exception")
+        }
+    }
+
+    fun createAndRunBackgroundEngine(context: Context, loader: FlutterLoader ){
+
+        // Get the background engine from FlutterEngineCache, returns null if not found
+        backgroundFlutterEngine = FlutterEngineCache.getInstance()
+            .get(PushNotificationPluginConstants.BACKGROUND_ENGINE_ID)
+
+        // If the Flutter engine is not found in cache, create and put into cache
+        if (backgroundFlutterEngine == null) {
+            // Create a background Flutter Engine
+            backgroundFlutterEngine = FlutterEngine(context)
+            // Put it into cache so the next notification coming through in killed state
+            // can use the same Flutter engine.
+            FlutterEngineCache.getInstance().put(
+                PushNotificationPluginConstants.BACKGROUND_ENGINE_ID,
+                backgroundFlutterEngine,
+            )
+        }
+        val callbackHandle = context.getSharedPreferences(
+            PushNotificationPluginConstants.SHARED_PREFERENCES_KEY,
+            Context.MODE_PRIVATE
+        ).getLong(
+            PushNotificationPluginConstants.BACKGROUND_FUNCTION_KEY, 0
+        )
+        if (callbackHandle == 0L) {
+//            Log.w(
```

Comment:
Remove or uncomment

Reply:
Resolved in this [open PR](https://github.com/aws-amplify/amplify-flutter/pull/2824)